src/tests/performance/connections.rs

```diff
@@ -0,0 +1,217 @@
+const METRIC_ERROR: &str = "perf_conn_error";
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+async fn p002_connections_load() {
+    // ZG-PERFORMANCE-002
+    //
+    // The node sheds or rejects connections when necessary.
+    //
+    //  1. Start a node with max_peers set to `N`
+    //  2. Initiate connections from `M > N` peer nodes
+    //  3. Expect only `N` to be active at a time
+    //
+    // Seems rippled not perform like the above way. max_peers is not a limit for connection
+    // number which is set (hardcoded?) to about 20 at the time. max_peer seems to make difference
+    // if any connections will be terminated. Need to investigate in the next commit.
+    //
+    // Sample result:
+    // ┌─────────────┬─────────┬──────────────┬──────────────┬──────────────┬──────────────┬──────────────┬────────────┐
+    // │             │         │  connection  │  connection  │  connection  │  connection  │  connection  │            │
+    // │  max peers  │  peers  │  accepted    │  rejected    │  terminated  │  error       │  timed out   │  time (s)  │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │       1 │            1 │            0 │            0 │            0 │            0 │       0.20 │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │       5 │            5 │            0 │            0 │            0 │            0 │       0.74 │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │      10 │           10 │            0 │            0 │            0 │            0 │       1.83 │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │      20 │           19 │            1 │            0 │            0 │            0 │       3.86 │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │      30 │           21 │            9 │           14 │            0 │            0 │       5.81 │
+    // ├─────────────┼─────────┼──────────────┼──────────────┼──────────────┼──────────────┼──────────────┼────────────┤
+    // │          50 │      50 │           20 │           30 │           19 │            0 │            0 │       9.53 │
+    // └─────────────┴─────────┴──────────────┴──────────────┴──────────────┴──────────────┴──────────────┴────────────┘
+
+    // maximum time allowed for a single iteration of the test
+    const MAX_ITER_TIME: Duration = Duration::from_secs(20);
+
+    /// maximum peers to configure node with
+    const MAX_PEERS: u16 = 50;
+
+    let synth_counts = vec![1u16, 5u16, 10u16, 20u16, 30u16, 50u16];

```

Comment:
to make it less noisy, only the first element can be used to define the type, so we could do this if you like:
```suggestion
    let synth_counts = vec![1u16, 5, 10, 20, 30, 50];

```

Reply:
Done.